app/src/processing/app/Editor.java

```diff
@@ -2280,22 +2280,51 @@ protected boolean checkModified() {
     // With Java 7u40 on OS X, need to bring the window forward.
     toFront();
     
-    String prompt = "Save changes to " + sketch.getName() + "?  ";
+    
 
     if (!Base.isMacOS()) {
-      int result =
-        JOptionPane.showConfirmDialog(this, prompt, "Close",
-                                      JOptionPane.YES_NO_CANCEL_OPTION,
-                                      JOptionPane.QUESTION_MESSAGE);
+      Object[] options = new String[] {
+        Language.text("save.btn.save"), Language.text("save.btn.dont_save"), Language.text("prompt.cancel")
+      };
+      JOptionPane optionPane = new JOptionPane("<html> " +
+        "<head> <style type=\"text/css\">"+
+        "b { font: 13pt \"Lucida Grande\" }"+
+        "p { font: 11pt \"Lucida Grande\"; margin-top: 8px }"+
+        "</style> </head>" +
+        "<b>" + Language.text("save.title") + "</b>" +
+        "<p>" + Language.text("save.hint") + "</p>", 
+              JOptionPane.QUESTION_MESSAGE, 
+              JOptionPane.YES_NO_CANCEL_OPTION,
+              null,
+              options);
+              
 
-      if (result == JOptionPane.YES_OPTION) {
+      JDialog dialog = optionPane.createDialog("Confirm Dialog");
+
+      Set<AWTKeyStroke> forwardTraversalKeys = 
```

Comment:
I am sorry, i should have added this with the original issue - 
Java does not proides the arrow bindings, it provides bindings with the `Tab` (which is by default)  
Though binding by `Tab` is enough i just thought it would be a "cool" feature as not many editors do that and i thought it can be useful.  
If you feel that this should be done i can extend JOptionPane and create a separate class for it.